components/brave_new_tab_ui/containers/newTab/index.tsx

```diff
@@ -179,10 +188,61 @@ class NewTabPage extends React.Component<Props, State> {
     )
   }
 
+  toggleShowCrypto = () => {
+    const { currentStackWidget } = this.props.newTabData
+
+    if (currentStackWidget === 'rewards') {
+      this.toggleShowRewards()
+    } else {
+      this.toggleShowBinance()
+    }
+  }
+
   toggleShowRewards = () => {
-    this.props.saveShowRewards(
-      !this.props.newTabData.showRewards
-    )
+    const {
+      currentStackWidget,
+      showBinance,
+      showRewards
+    } = this.props.newTabData
+
+    if (currentStackWidget === 'rewards') {
+      this.props.actions.setCurrentStackWidget('binance')
+    } else if (!showBinance) {
+      this.props.actions.setCurrentStackWidget('rewards')
+    }
+
+    this.props.saveShowRewards(!showRewards)
+  }
+
+  toggleShowBinance = () => {
+    const {
+      currentStackWidget,
+      showBinance,
+      showRewards
+    } = this.props.newTabData
+
+    if (currentStackWidget === 'binance') {
+      this.props.actions.setCurrentStackWidget('rewards')
+    } else if (!showRewards) {
+      this.props.actions.setCurrentStackWidget('binance')
+    }
+
+    this.props.saveShowBinance(!showBinance)
+  }
+
+  buyCrypto = (coin: string, amount: string, fiat: string) => {
+    const { userTLD } = this.props.newTabData.binanceState
+    const refParams = 'ref=39346846&utm_source=brave'
+
+    if (userTLD === 'us') {
+      window.open(`https://www.binance.${userTLD}/en/buy-sell-crypto?coin=${coin}&amount=${amount}&${refParams}`, '_blank')
```

Comment:
See comment in https://github.com/brave/brave-core/pull/4910/files#diff-018be6619d60f19c79a62f9b09840db8R234